src/web/View.php

```diff
@@ -643,6 +649,48 @@ public function clearJsBuffer(bool $scriptTag = true)
     }
 
     /**
+     * Registers a generic <script> code block.
+     * @param string $script the generic <script> code block to be registered
+     *
+     * @param string $key the key that identifies the generic <script> code block. If null, it will use
+     * $script as the key. If two generic <script> code blocks are registered with the same key, the latter
+     * will overwrite the former.
+     */
+    public function registerScript($script, $options = [], $key = null)
+    {
+        $key = $key ?: md5($script);
+        $this->_scripts[$key] = Html::script($script, $options);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function renderHeadHtml()
+    {
+        $lines = [];
+        if (!empty($this->title)) {
+            $lines[] = '<title>' . Html::encode($this->title) . '</title>' . "\n";
+        }
+
+        $html = parent::renderHeadHtml();
+        return empty($lines) ? $html : implode("\n", $lines) . $html;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function renderBodyEndHtml($ajaxMode)
+    {
+        $lines = [];
+        if (!empty($this->_scripts)) {
+            $lines[] = implode("\n", $this->_scripts);
```

Comment:
This would result in

```php
$lines = [
    [
        '<script ... />',
        // ...
    ]
];
```

rather than

```php
$lines = [
    '<script ... />',
    // ...
];
```